.github/workflows/release.yml

```diff
@@ -22,33 +25,208 @@ jobs:
+          GITHUB_ENV: ${{ env.GITHUB_ENV }}
+
+      - name: parse env (for scheduled runs)
+        if: github.event_name != 'workflow_dispatch' && steps.check_week.outputs.RUN_WORKFLOW == 'true'
         run: |
-          echo "VERSION=$(echo "${{inputs.tag_name}}" | cut -d '-' -f1)" >> "$GITHUB_ENV"
+          current_tag="${{ steps.get_latest_tag.outputs.LATEST_TAG }}"
+          current_version=$(echo "${current_tag}" | cut -d'.' -f2)
+          next_version=$((current_version + 1))
+          new_tag="v0.${next_version}.0"
+          echo "VERSION=${new_tag}" >> "$GITHUB_ENV"
+          echo "PREVIOUS_TAG=${current_tag}" >> "$GITHUB_ENV"
+          echo "Print env:VERSION and env:PREVIOUS_TAG"
+          echo "VERSION=${new_tag}"
+          echo "PREVIOUS_TAG=${current_tag}"
       - name: generate changelog
+        if: steps.check_week.outputs.RUN_WORKFLOW == 'true'
         run: |
           set -e
+          target_commitish="${{ github.event.inputs.target_commitish || 'main' }}"
           response=$(curl -L \
           -X POST \
           -H "Accept: application/vnd.github+json" \
           -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
           -H "X-GitHub-Api-Version: 2022-11-28" \
           https://api.github.com/repos/Edgenesis/shifu/releases/generate-notes \
-          -d '{"tag_name":"${{inputs.tag_name}}","target_commitish":"${{inputs.target_commitish}}","previous_tag_name":"${{inputs.previous_tag_name}}"}')
+          -d '{"tag_name":"${{ env.NEW_TAG }}","target_commitish":"'"${target_commitish}"'","previous_tag_name":"${{ env.PREVIOUS_TAG }}"}')
           echo $response
           go run tools/release/release.go "$response"
         env:
           AZURE_OPENAI_APIKEY: ${{ secrets.AZURE_OPENAI_APIKEY }}
           AZURE_OPENAI_HOST: ${{ secrets.AZURE_OPENAI_HOST }}
           DEPLOYMENT_NAME: ${{secrets.DEPLOYMENT_NAME}}
+          GITHUB_ENV: ${{ env.GITHUB_ENV }}
+
       - name: Create Pull Request
+        if: steps.check_week.outputs.RUN_WORKFLOW == 'true'
         uses: peter-evans/create-pull-request@v5
         with:
-          commit-message: add changelog for ${{env.VERSION}}
-          title: add changelog for ${{env.VERSION}}
-          body: add changelog for ${{env.VERSION}}
-          branch: changelog-${{env.VERSION}}
+          commit-message: add changelog for ${{ env.NEW_TAG }}
+          title: add changelog for ${{ env.NEW_TAG }}
+          body: add changelog for ${{ env.NEW_TAG }}
+          branch: changelog-${{ env.NEW_TAG }}
           base: main
+
+      - name: Fetch Pull Request Number
+        if: steps.check_week.outputs.RUN_WORKFLOW == 'true'
+        id: get-pr
+        run: |
+          PR_NUMBER=$(gh pr list --base main --head changelog-${{ env.NEW_TAG }} --json number --jq '.[0].number')
+          echo $PR_NUMBER
+          echo "PR_NUMBER=${PR_NUMBER}" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Merge Pull Request
```

Comment:
merging after approval?

Reply:
It automatically merges. Is approval needed?